lib/io.js

```diff
@@ -123,7 +123,17 @@ module.exports = app => {
   }
 
   if (config.redis) {
-    app.io.adapter(redis(config.redis));
+    // Use ioredis to support sentinel mode if redis config
+    // has 'sentinels' item.
+    if (config.redis.sentinels) {
```

Comment:
egg-redis is optionalDependencies, so we should do the assert here.

assert(app.redis, '[egg-socket.io] egg-redis must be enabled when sentinels is configured');